src/core/Darwin.zig

```diff
@@ -76,193 +59,299 @@ pub fn run(comptime on_each_update_fn: anytype, args_tuple: std.meta.ArgsTuple(@
     // TODO: support UIKit.
 }
 
-pub fn init(
-    darwin: *Darwin,
+pub fn tick(core: *Core) !void {
+    var windows = core.windows.slice();
+    while (windows.next()) |window_id| {
+        if (core.windows.getAll(window_id)) |cw| {
+            const core_window = cw;
+            const native: Native = core_window.native;
+
+            if (core.windows.get(window_id, .device).? == null) {
+                // Window has not been set up yet, we need to init the window
+                //try initWindow(core, window_id);
+                //try core.initWindow(window_id);
+
+                try initWindow(core, window_id);
+                try core.initWindow(window_id);
+            }
+            if (native.window) |native_window| {
```

Comment:
Ohhh, you know what we should do: let's use the `native` field as the optional value indicating 'the window was initialized'. Then the `.device` field can be `undefined` prior to that rather than nullable